### ISSUE TITLE
Incorrect parameter update logic in _cocob.py

### ISSUE THREAD

**Issue by ajitashwathr10** [google-deepmind/optax#1227]
In the Optax implementation of the COCOB optimizer, I found a fundamental flaw in the parameter update logic that conflicts with both the original algorithm and the way Optax's transformation system is intended to operate. In Optax, gradient updates should yield values to be used as updates that will be applied to parameters ``(optax.apply_updates())``, and not replacement parameter values. The present implementation tries to calculate the new parameter values themselves directly with an equation that begins with (-p + (.)) instead of calculating the changes that must be applied to the existing parameters.

```python
new_updates = jax.tree.map(
    lambda p, p0, C, L, G, R: (
        -p + (p0 + C / (L * jnp.maximum(G + L, alpha * L)) * (L + R))
    ),
    params,
    init_particles,
    cumulative_grads,
    scale,
    subgradients,
    reward,
)
```
